src/kubeobjects/secret.go

```diff
@@ -70,6 +84,42 @@ func (query SecretQuery) CreateOrUpdate(secret corev1.Secret) error {
 	return nil
 }
 
+func (query SecretQuery) CreateOrUpdateForNamespacesList(newSecret corev1.Secret, namespaces []corev1.Namespace) (int, error) {
+	secretList, err := query.GetAllFromNamespaces(newSecret.Name)
+	if err != nil {
+		return 0, err
+	}
+
+	namespacesContainingSecret := make(map[string]corev1.Secret)
```

Comment:
since we know the length the map is going to have (`len(secretList)`) we could allocate that directly